src/pages/VenuePage/TemplateWrapper.tsx

```diff
@@ -143,16 +145,25 @@ const TemplateWrapper: React.FC<TemplateWrapperProps> = ({ venue }) => {
       template = <div>Unknown Template: ${(venue as AnyVenue).template}</div>;
   }
 
-  // @debt remove backButton from Navbar
+  const parentVenueId = venue?.parentId ?? "";
+
+  const backToParentVenue = useCallback(() => {
+    enterVenue(parentVenueId, { customOpenRelativeUrl: history.push });
+  }, [history.push, parentVenueId]);
+
   return (
     <RelatedVenuesProvider venueId={venue.id}>
       <ReactionsProvider venueId={venue.id}>
-        <WithNavigationBar
-          fullscreen={fullscreen}
-          hasBackButton={hasBackButton}
-        >
+        <WithNavigationBar fullscreen={fullscreen}>
           <AnnouncementMessage message={venue.bannerMessage} />
           {template}
+          {!TemplatesWithoutBackButton.includes(venue.template) &&
+            parentVenue && (
```

Comment:
Extract this into a `const` outside of the JSX called `showBackButton` or `shouldShowBackButton` or similar.